### PR TITLE
STY: use pytest.raises context syntax (groupby)

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #24332

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def get_converter(s):
                     return [self.indices[name] for name in names]
                 except KeyError:
                     # turns out it wasn't a tuple
-                    msg = ("must supply a a same-length tuple to get_group"
+                    msg = ("must supply a same-length tuple to get_group"
```
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py]: be aware that I have changed an error message in the codebase

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -823,12 +840,22 @@ def test_groupby_level_nonmulti():
...
+        s.groupby(level=-2)
+    msg = "No group keys passed!"
+    with pytest.raises(ValueError, match=msg):
+        s.groupby(level=[])
+    msg = "multiple levels only valid with MultiIndex"
+    with pytest.raises(ValueError, match=msg):
+        s.groupby(level=[0, 0])
+    with pytest.raises(ValueError, match=msg):
+        s.groupby(level=[0, 1])
+    msg = "level > 0 or level < -1 only valid with  MultiIndex"
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_groupby.py]: i should probably remove the double space as well?

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -722,8 +734,11 @@ def test_omit_nuisance(df):
 
     # won't work with axis = 1
     grouped = df.groupby({'A': 0, 'C': 0, 'D': 1, 'E': 1}, axis=1)
-    result = pytest.raises(TypeError, grouped.agg,
-                           lambda x: x.sum(0, numeric_only=False))
+    msg = (r'\("unsupported operand type\(s\) for \+: '
+           "'Timestamp' and 'float'"
+           '"'r", 'occurred at index 0'\)")
```
Comment (WillAyd) [inline pandas/tests/groupby/test_groupby.py]: This is kind of tough to read - can you move the double quote to the end of the line before and have this one start with the `r` prefix?

File: pandas/tests/groupby/test_grouping.py [hunk 4]
```diff
@@ -26,9 +26,9 @@ class TestSelection(object):
     def test_select_bad_cols(self):
         df = DataFrame([[1, 2]], columns=['A', 'B'])
         g = df.groupby('A')
-        pytest.raises(KeyError, g.__getitem__, ['C'])  # g[['C']]
+        with pytest.raises(KeyError, match='"Columns not found: '"'C'"'"'):
```
Comment (WillAyd) [inline pandas/tests/groupby/test_grouping.py]: Also tough to read with quoting - can you not just escape what you need to?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor stylistic nits but generally looks good

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (WillAyd) [reply pandas/tests/groupby/test_groupby.py]: Good catch and yes that would be good

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @simonjayhawkins

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (gfyoung) [reply pandas/core/groupby/groupby.py]: And this is why we should be checking error messages if possible, to catch things like this 🙂

**Pull request merged**